snapcraft/tests/plugins/test_python.py

```diff
@@ -86,7 +211,93 @@ class Options:
         self.addCleanup(patcher.stop)
 
 
-class PythonPluginTestCase(BasePythonPluginTestCase):
+class PythonPluginPullTestCase(BasePythonPluginTestCase):
+
+    def test_with_unexisting_setup_file(self):
+        self.options.python_packages = ['dummy']
+        plugin = python.PythonPlugin('test-part', self.options,
+                                     self.project_options)
+        setup_directories(
+            plugin, self.options.python_version, setup_file=False)
+        with FakePip() as fake_pip:
+            plugin.pull()
+            # XXX Just a safeguard to make sure that the mock doesn't get out
+            # of sync.
+            self.assertThat(
+                fake_pip.mock_call.mock_calls[2][1][0],
+                Not(Contains('.')))
+
+    def test_with_setup_file(self):
+        self.options.python_packages = ['dummy']
+        plugin = python.PythonPlugin('test-part', self.options,
+                                     self.project_options)
+        setup_directories(plugin, self.options.python_version, setup_file=True)
+        with FakePip() as fake_pip:
+            plugin.pull()
+            # XXX Just a safeguard to make sure that the mock doesn't get out
+            # of sync.
+            self.assertThat(
+                fake_pip.mock_call.mock_calls[2][1][0][-2],
+                Equals('.'))
+
+    def test_pull_with_nothing(self):
+        plugin = python.PythonPlugin('test-part', self.options,
+                                     self.project_options)
+        setup_directories(plugin, self.options.python_version)
+
+        with FakePip() as fake_pip:
+            plugin.pull()
+            self.assertThat(fake_pip.download['constraints'], Equals([]))
+            self.assertThat(fake_pip.download['requirements'], Equals([]))
+            self.assertThat(fake_pip.download['downloaded'], Equals([]))
```

Comment:
just by reading this and not the fake, this check really doesn't make sense given the key names used here.

Reply:
I've added a get_downloaded method to the fake. Please take a look and let me know if it's better.